profile-widget/profilescene.cpp

```diff
@@ -45,23 +45,42 @@ class ProfileAnimation : public QAbstractAnimation {
 };
 
 template<typename T, class... Args>
-T *ProfileScene::createItem(const DiveCartesianAxis &vAxis, int vColumn, int z, Args&&... args)
+T *ProfileScene::createItem(const DiveCartesianAxis &vAxis, DataAccessor accessor, int z, Args&&... args)
 {
-	T *res = new T(*dataModel, *timeAxis, vAxis, vColumn, std::forward<Args>(args)...);
+	T *res = new T(*dataModel, *timeAxis, vAxis, accessor, std::forward<Args>(args)...);
 	res->setZValue(static_cast<double>(z));
 	profileItems.push_back(res);
 	return res;
 }
 
-PartialPressureGasItem *ProfileScene::createPPGas(int column, color_index_t color, color_index_t colorAlert,
-						    const double *thresholdSettingsMin, const double *thresholdSettingsMax)
+PartialPressureGasItem *ProfileScene::createPPGas(DataAccessor accessor, color_index_t color, color_index_t colorAlert,
+						  const double *thresholdSettingsMin, const double *thresholdSettingsMax)
 {
-	PartialPressureGasItem *item = createItem<PartialPressureGasItem>(*gasYAxis, column, 99, dpr);
+	PartialPressureGasItem *item = createItem<PartialPressureGasItem>(*gasYAxis, accessor, 99, dpr);
 	item->setThresholdSettingsKey(thresholdSettingsMin, thresholdSettingsMax);
 	item->setColors(getColor(color, isGrayscale), getColor(colorAlert, isGrayscale));
 	return item;
 }
 
+template <int IDX>
+double accessTissue(const plot_data &item)
+{
+	return item.ceilings[IDX];
+}
+
+// For now, the accessor functions for the profile data do not possess a payload.
+// To generate the 16 tissue (ceiling) accessor functions, use iterative templates.
+// Thanks to C++17's constexpr if, this is actually easy to read and follow.
```

Comment:
"_pleasantly easy to follow..._"
Right.
I have absolutely not the faintest clue what this code does.
If you ever decide that you are done with Subsurface and move on, I have no idea who will maintain this code, because it certainly ain't me.
That's not to say "don't do that". It's to say "don't stop working on Subsurface, like, ever"...

Reply:
> I have absolutely not the faintest clue what this code does.

Actually, I'm confident that you will figure it out in no time if you ever have to. `if constexpr (...)` is executed in the compiler and this implements a simple loop counting from 0 to 15 to generate the 16 accessor functions.

It's well possible that we'll replace this with a more dynamic approach (e.g. virtual functions) if it turns out that this is too restrictive.